lib/internal/main/worker_thread.js

```diff
@@ -163,6 +165,29 @@ process._fatalException = (error) => {
   }
 };
 
+function makeEnvProxy(env) {
+  return new Proxy(env, {
+    set(target, key, value) {
+      // Override 'set' to make sure only strings are stored in our copy of
+      // `process.env`. The deprecation warning text and behaviour matches
+      // what the main thread does in C++.
+      if (typeof value !== 'string' &&
+          typeof value !== 'number' &&
+          typeof value !== 'boolean') {
+        const { getOptionValue } = require('internal/options');
+        if (getOptionValue('--pending-deprecation')) {
+          process.emitWarning(
+            'Assigning any value other than a string, number, or boolean to ' +
+            'a process.env property is deprecated. Please make sure to ' +
+            'convert the value to a string before setting process.env with it.',
+            'DeprecationWarning', 'DEP0104');
+        }
+      }
+      return Reflect.set(target, key, `${value}`);
+    }
```

Comment:
We should likely also add this for `Object.defineProperty()`.

Reply:
[MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/handler/defineProperty) says:

> Note that this trap will not be entered if a `handler.set()` method has been defined.

But that does not appear to be true? Sigh.

Reply:
This seems to be a mistake in MDN. See [spec part 9.5.6](https://tc39.github.io/ecma262/#sec-proxy-object-internal-methods-and-internal-slots-defineownproperty-p-desc) step 7.